src/accountExporter.ts

```diff
@@ -0,0 +1,211 @@
+import { googleOrigin } from "./api";
+import * as utils from "./utils";
+
+const apiClient = new Client({
+  urlPrefix: googleOrigin,
+});
+
+// TODO: support for MFA at runtime was added in PR #3173, but this exporter currently ignores `mfaInfo` and loses the data on export.
+const EXPORTED_JSON_KEYS = [
+  "localId",
+  "email",
+  "emailVerified",
+  "passwordHash",
+  "salt",
+  "displayName",
+  "photoUrl",
+  "lastLoginAt",
+  "createdAt",
+  "phoneNumber",
+  "disabled",
+  "customAttributes",
+];
+const EXPORTED_JSON_KEYS_RENAMING: Record<string, string> = {
+  lastLoginAt: "lastSignedInAt",
+};
+const EXPORTED_PROVIDER_USER_INFO_KEYS = [
+  "providerId",
+  "rawId",
+  "email",
+  "displayName",
+  "photoUrl",
+];
+const PROVIDER_ID_INDEX_MAP: Record<string, number> = {
+  "google.com": 7,
+  "facebook.com": 11,
+  "twitter.com": 15,
+  "github.com": 19,
+};
+
+function escapeComma(str: string): string {
+  if (str.includes(",")) {
+    // Encapsulate the string with quotes if it contains a comma.
+    return `"${str}"`;
+  }
+  return str;
+}
+
+function convertToNormalBase64(data: string): string {
+  return data.replace(/_/g, "/").replace(/-/g, "+");
+}
+
+function addProviderUserInfo(providerInfo: any, arr: any[], startPos: number): void {
+  arr[startPos] = providerInfo.rawId;
+  arr[startPos + 1] = providerInfo.email || "";
+  arr[startPos + 2] = escapeComma(providerInfo.displayName || "");
+  arr[startPos + 3] = providerInfo.photoUrl || "";
+}
+
+function transUserToArray(user: any): any[] {
+  const arr = Array(27).fill("");
+  arr[0] = user.localId;
+  arr[1] = user.email || "";
+  arr[2] = user.emailVerified || false;
+  arr[3] = convertToNormalBase64(user.passwordHash || "");
+  arr[4] = convertToNormalBase64(user.salt || "");
+  arr[5] = escapeComma(user.displayName || "");
+  arr[6] = user.photoUrl || "";
+  for (let i = 0; i < (!user.providerUserInfo ? 0 : user.providerUserInfo.length); i++) {
+    const providerInfo = user.providerUserInfo[i];
+    if (providerInfo && PROVIDER_ID_INDEX_MAP[providerInfo.providerId]) {
+      addProviderUserInfo(providerInfo, arr, PROVIDER_ID_INDEX_MAP[providerInfo.providerId]);
+    }
+  }
+  arr[23] = user.createdAt;
+  arr[24] = user.lastLoginAt;
+  arr[25] = user.phoneNumber;
+  arr[26] = user.disabled;
+  arr[27] = user.customAttributes;
+  return arr;
+}
+
+function transUserJson(user: any): any {
+  const newUser: any = {};
+  _.each(_.pick(user, EXPORTED_JSON_KEYS), (value, key) => {
+    const newKey = EXPORTED_JSON_KEYS_RENAMING[key] || key;
+    newUser[newKey] = value;
+  });
+  if (newUser.passwordHash) {
+    newUser.passwordHash = convertToNormalBase64(newUser.passwordHash);
+  }
+  if (newUser.salt) {
+    newUser.salt = convertToNormalBase64(newUser.salt);
+  }
+  if (user.providerUserInfo) {
+    newUser.providerUserInfo = [];
+    user.providerUserInfo.forEach((providerInfo: any) => {
+      if (!_.includes(Object.keys(PROVIDER_ID_INDEX_MAP), providerInfo.providerId)) {
```

Comment:
This seems like a much more convoluted way of writing:
if (PROVIDER_ID_INDEX_MAP[providerInfo.providerId] === undefined) {
}